### ISSUE TITLE
`exif_read` error when ExifTool doesn't find any files to read in recursive mode

### ISSUE THREAD

**Issue by jcblum** [JoshOBrien/exiftoolr#20]
I encountered this when running `exif_read()` recursively with a file extension filter that happened to exclude all possible files, but I think more generally this is about the `system2()` call to ExifTool returning `character(0)`, which triggers a `jsonlite::fromJSON()` error. 

ExifTool itself doesn't consider this situation to be an error (on the command line, it would report that it read 0 files but otherwise exit with status `0`).

``` r
library(exiftoolr)

exif_read(
  path = system.file(package = "exiftoolr", "images"),
  recursive = TRUE,
  tags = "filename", 
  args = c("--ext", "jpg")
)
#> Error: parse error: premature EOF
#>                                        
#>                      (right here) ------^
```

<sup>Created on 2023-11-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessionInfo()
#> R version 4.3.2 (2023-10-31)
#> Platform: aarch64-apple-darwin20 (64-bit)
#> Running under: macOS Monterey 12.6.6
#> 
#> Matrix products: default
#> BLAS:   /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRblas.0.dylib 
#> LAPACK: /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.11.0
#> 
#> locale:
#> [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
#> 
#> time zone: America/Los_Angeles
#> tzcode source: internal
#> 
#> attached base packages:
#> [1] stats     graphics  grDevices utils     datasets  methods   base     
#> 
#> other attached packages:
#> [1] exiftoolr_0.2.3
#> 
#> loaded via a namespace (and not attached):
#>  [1] vctrs_0.6.4       zip_2.3.0         cli_3.6.1         knitr_1.45       
#>  [5] rlang_1.1.2       xfun_0.41         purrr_1.0.2       styler_1.10.2    
#>  [9] jsonlite_1.8.7    data.table_1.14.8 glue_1.6.2        backports_1.4.1  
#> [13] htmltools_0.5.7   rmarkdown_2.25    R.cache_0.16.0    evaluate_0.23    
#> [17] fastmap_1.1.1     yaml_2.3.7        lifecycle_1.0.4   compiler_4.3.2   
#> [21] fs_1.6.3          rstudioapi_0.15.0 R.oo_1.25.0       R.utils_2.12.2   
#> [25] digest_0.6.33     reprex_2.0.2      curl_5.1.0        magrittr_2.0.3   
#> [29] R.methodsS3_1.8.2 tools_4.3.2       withr_2.5.2
```

</details>

I really appreciate the time you've put into this super useful package! I'd be happy to contribute a PR to help out with a fix for this issue, if you have thoughts about how you'd want to handle it: warning (with a more informative message), return an empty _something_, or some other alternative?

As a side note, ExifTool _does_ consider it an error if you've passed one or more file paths non-recursively and the extension filter excluded all of them. `exif_read()` is also erroring in that scenario, but only because `fromJSON()` still fails. `system2` reports a warning, but not an error (and I suspect the warning may be a bit inscrutable for users who don't know how `exif_call()` works internally).

<details style="margin-bottom:10px;">
<summary>
Non-recursive example
</summary>

``` r
library(exiftoolr)

exif_read(
  path = system.file(package = "exiftoolr", "images", "Lizard.jpg"), 
  tags = "filename", 
  args = c("--ext", "jpg")
)
#> Warning in system2(exiftoolpath, args = args, stdout = stdout): running command
#> ''exiftool' -@
#> '/var/folders/2y/jjlk1tgx0tngz12f48g3znlr0000gr/T//RtmpHKhseC/args.cmd96cf7b64506e''
#> had status 1
#> Error: parse error: premature EOF
#>                                        
#>                      (right here) ------^

exif_read(
  path = c(system.file(package = "exiftoolr", "images", "LaSals.jpg"), 
           system.file(package = "exiftoolr", "images", "Lizard.jpg")), 
  tags = "filename", 
  args = c("--ext", "jpg")
)
#> Warning in system2(exiftoolpath, args = args, stdout = stdout): running command
#> ''exiftool' -@
#> '/var/folders/2y/jjlk1tgx0tngz12f48g3znlr0000gr/T//RtmpHKhseC/args.cmd96cf76dca50b''
#> had status 1
#> Error: parse error: premature EOF
#>                                        
#>                      (right here) ------^
```

</details>

**Comment by JoshOBrien** [comment]
@jcblum Thanks for taking the time to report this issue, and for so clearly describing its genesis.

I went ahead and added one possible fix to `exif_read()`, which: (1) catches and passes on any ExifTool error or warning message as an R warning; and (2) returns a value of `NULL`. You can test that out if you would like by doing:

```r
install_github("JoshOBrien/exiftoolr", ref = "handle-zero-file-results")
```

Let me know what you think, and if you have any suggestions for a better way to handle the results in cases like the ones you supplied, I'd for sure welcome them.

Cheers,

- Josh

**Comment by JoshOBrien** [comment]
Closed with ce063c5480dd797172bbf299ece4a4b7d2baed29, now on its way to CRAN.
